From aaf0e610d839d82abde1da45d8ead9adbb71d2cb Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Sat, 25 Mar 2017 05:35:26 +0000
Subject: [PATCH] s390x: provide a working sigcontext struct definition

For some reason, struct sigcontext was not declared on s390x (even though it is declared on every
other architecture).  Provide a workable equivalent to the structure provided by the kernel.

Signed-off-by: William Pitcock <nenolod@dereferenced.org>
---
 arch/s390x/bits/signal.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/s390x/bits/signal.h b/arch/s390x/bits/signal.h
index c8665832..832351ce 100644
--- a/arch/s390x/bits/signal.h
+++ b/arch/s390x/bits/signal.h
@@ -43,6 +43,11 @@ typedef struct {
 
 #endif
 
+struct sigcontext {
+	unsigned long oldmask[1];
+	mcontext_t *sregs;
+};
+
 struct sigaltstack {
 	void *ss_sp;
 	int ss_flags;
-- 
2.12.1

